uc-crux-llvm/src/UCCrux/LLVM/Newtypes/PreSimulationMem.hs

```diff
@@ -0,0 +1,23 @@
+{-
+Module           : UCCrux.LLVM.Newtypes.PreSimulationMem
+Copyright        : (c) Galois, Inc 2020
+License          : BSD3
+Maintainer       : Langston Barrett <langston@galois.com>
+Stability        : provisional
+-}
+
+module UCCrux.LLVM.Newtypes.PreSimulationMem
+  ( PreSimulationMem
+  , makePreSimulationMem
+  , getPreSimulationMem
+  ) where
+
+import           Lang.Crucible.LLVM.MemModel (MemImpl)
+
+-- | LLVM memory just before symbolic execution. Populated with globals, defined
+-- functions, and arguments to the entry point.
```

Comment:
I like newtypes, but what is the motivation? Is it that we want to inspect pre-states sometimes and want to make sure that we don't mix up memimpls?

Reply:
Yeah, pretty much! It's to distinguish this kind of use from other `MemImpl`s, hopefully making it easier to understand what's going on, and specifically, what sorts of reads one would expect to work on this particular `MemImpl`.